cloud/kyup/kyup.py

```diff
@@ -0,0 +1,338 @@
+            return ret
+        else:
+            if 'data' in ret and 'error_code' in ret['data']:
+                if ret['data']['error_code'] == 101:
+                    module.fail_json(changed=False, msg = 'Failed to execute request. Error code: %d Error msg: %s req: %s' % (ret['data']['error_code'], ret['data']['error'], data))
+                else:
+                    module.fail_json(changed=False, msg = 'Failed to execute request. Error code: %d Error msg: %s' % (ret['data']['error_code'], ret['data']['error']))
+    else:
+        module.fail_json(changed=False, msg = 'Req: ' + data + ' Resp: ' + ret)
+
+def kyup_action(module, action, container_id = 0):
+    req = '{"action":"cloudList","authorization_key":"%s","data":{}}'
+    ret = api_request(module, req % module.params.get('api_key'))
+    for container in ret['data']['list']:
+        if container['name'] == module.params.get('name'):
+          container_id = container['id']
+          break
+
+    if container_id == 0:
+        module.exit_json(changed=False, msg = 'Could not find container id for container name %s' % module.params.get('name'))
+
+    req = '{"action":"%s","authorization_key":"%s","data":{"container_id":%d}}'
+    ret = api_request(module, req % (action, module.params.get('api_key'), int(container_id)) )
+    if action == 'cloudDetails':
+        return ret['data']['container']
+    else:
+        module.exit_json(changed=True, msg = '%s successfull' % action)
+
+def get_task_status(module, task_id):
+    req = '{"action":"getTask","authorization_key":"%s","data":{"task_id":%d}}'
+    count = retries;
+    while count != 0:
+        ret = api_request(module, req % (module.params.get('api_key'), int(task_id)) )
+        if 'container_id' in ret['data']['task']:
+            return ret['data']['task']['container_id']
+        count -= 1
+    module.fail_json(changed=False, msg = ret)
+
+def add_ssh_keys(module, container_id):
+    key_list = {}
+    ssh_keys = module.params['ssh_keys']
+    if ssh_keys is None or ssh_keys == '':
+        return
+    # cleanup any spaces put by the user
+    ssh_keys.replace(' ', '')
+
+    for i in ssh_keys.split(','):
+        key_list[i] = 0
+    # get the key IDs
+    req = '{"action":"%s","authorization_key":"%s","data":{%s}}'
+    ret = api_request(module, req % ('sshGetKeys', module.params.get('api_key'), ''))
+    for key in ret['data']['keys']:
+        if key['title'] in key_list:
+            key_list[key['title']] = key['key_id']
+    # add the keys to the container
+    key_req = '"key_id":%d,"container_id":%d'
+    for key in key_list:
+        if key_list[key] == 0:
+            module.fail_json(changed=True, msg = 'Unable to find key ID for key %s' % key)
+        api_request(module, req % ('sshInstallKey', module.params.get('api_key'), key_req % (int(key_list[key]), int(container_id) )))
+
+def create_container(module):
+    enc_key = module.params['enc_key'] or os.environ['KYUP_ENC_KEY']
```

Comment:
If `module.params['enc_key']` is false and `KYUP_ENC_KEY` env var isn't set this will raise an KeyError exception.

`os.environ.get('KYUP_ENC_KEY', None)` will do the trick